File: backend/src/main/java/org/certificaciones/funproeibbackend/service/ProgramaService.java
package org.certificaciones.funproeibbackend.service;

import org.certificaciones.funproeibbackend.dto.ProgramaRequest;
import org.certificaciones.funproeibbackend.dto.ProgramaResponse;

import java.util.List;

public interface ProgramaService {
    ProgramaResponse crear(ProgramaRequest request);
    ProgramaResponse obtenerPorId(Long id);
    List<ProgramaResponse> listarTodos();
    ProgramaResponse actualizar(Long id, ProgramaRequest request);
    ProgramaResponse cambiarEstado(Long id, String nuevoEstado);
}

File: frontend/src/app/core/services/programa.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';
import { ProgramaRequest, ProgramaResponse } from '../models/programa.model';

@Injectable({ providedIn: 'root' })
export class ProgramaService {
  constructor(private api: ApiService) {}

  listarTodos(): Observable<ProgramaResponse[]> {
    return this.api.get<ProgramaResponse[]>('/programas');
  }

  obtenerPorId(id: number): Observable<ProgramaResponse> {
    return this.api.get<ProgramaResponse>(`/programas/${id}`);
  }

  crear(request: ProgramaRequest): Observable<ProgramaResponse> {
    return this.api.post<ProgramaResponse>('/programas', request);
  }

  actualizar(id: number, request: ProgramaRequest): Observable<ProgramaResponse> {
    return this.api.put<ProgramaResponse>(`/programas/${id}`, request);
  }

  cambiarEstado(id: number, estado: string): Observable<ProgramaResponse> {
    return this.api.patch<ProgramaResponse>(`/programas/${id}/estado`, { estado });
  }
}
